### PR TITLE
Upgrade to .NET Core and Bot Framework v4 SDK

### PR DESCRIPTION
**Pull request by adlele**
Updated sample to dot net core  and bot framework v4

### CODE REVIEW

File: Microsoft.Teams.Samples.HelloWorld.Web/Bots/MessageExtension.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                   
+            return Task.FromResult(new MessagingExtensionResponse
+            {
+                ComposeExtension = new MessagingExtensionResult
+                {
+                    AttachmentLayout = "list",
+                    Type = "result",
+                    Attachments = new MessagingExtensionAttachment[]{
+                        new ThumbnailCard()
+                                    .ToAttachment()
```
Comment (aosolis) [inline Microsoft.Teams.Samples.HelloWorld.Web/Bots/MessageExtension.cs:86]: Nit: Indent 4 spaces only

File: Microsoft.Teams.Samples.HelloWorld.Web/Controllers/BotController.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            Adapter = adapter;
+            Bot = bot;
+        }
+
+        [HttpPost, HttpGet]
+        public async Task PostAsync()
+        {
+            // Delegate the processing of the HTTP POST to the adapter.
+            // The adapter will invoke the bot.
+
```
Comment (aosolis) [inline Microsoft.Teams.Samples.HelloWorld.Web/Controllers/BotController.cs:34]: Nit: Remove blank line

File: Microsoft.Teams.Samples.HelloWorld.Web/Controllers/BotController.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        private readonly IBotFrameworkHttpAdapter Adapter;
+        private readonly IBot Bot;
+
+        public BotController(IBotFrameworkHttpAdapter adapter, IBot bot)
+        {
+            Adapter = adapter;
+            Bot = bot;
+        }
+
+        [HttpPost, HttpGet]
```
Comment (aosolis) [inline Microsoft.Teams.Samples.HelloWorld.Web/Controllers/BotController.cs:29]: HttpPost only-- this doesn't handle GET, does it?

File: Microsoft.Teams.Samples.HelloWorld.Web/Controllers/BotController.cs (hunk 2 above)
Comment (adlele) [reply Microsoft.Teams.Samples.HelloWorld.Web/Controllers/BotController.cs]: done

File: Microsoft.Teams.Samples.HelloWorld.Web/Controllers/BotController.cs (hunk 3 above)
Comment (adlele) [reply Microsoft.Teams.Samples.HelloWorld.Web/Controllers/BotController.cs]: no. should be only POST

**Pull request merged**